### ISSUE TITLE
npmPublish setting ignored

### ISSUE THREAD

**Issue by aaroncraigongithub** [semantic-release/npm#137]
Running the release skips publishing to npm because the `private` field is set to true, however, I have set `npmPublish: true` in `.releaserc`.

```json
// .releaserc
{
  "plugins": [
    "@semantic-release/commit-analyzer",
    "@semantic-release/release-notes-generator",
    "@semantic-release/gitlab",
    "@semantic-release/changelog",
    ["@semantic-release/npm", {
      "npmPublish": true
    }],
    "@semantic-release/git"
  ]
}
```

```bash
# job output
# [snip]
[19:47:08] [semantic-release] [@semantic-release/npm] › ℹ  Skip publishing to npm registry as package.json's private property is true
```

**Comment by aaroncraigongithub** [comment]
Looks like the logic here is incorrect? 
https://github.com/semantic-release/npm/blob/master/lib/publish.js#L16

I believe it should be: 
```javascript
if (npmPublish !== false && pkg.private !== true) {
```

**Comment by pvdlg** [comment]
This is intentional. We made the choice of publishing only when there is no doubt about the user intention. If those two properties contradict each other we skip the publication to be safe.


In any case, what you try to achieve will not work, even with your change.
With what you propose, semantic-release would call `npm publish` and it will errors out because the `private` property of `package.json` is `true`.

If you want to publish your module on the npm repository, don't set it to `private`.

**Comment by travi** [comment]
@aaroncraigongithub are you maybe wanting `restricted` for `publishConfig.access` instead of `private`?

**Comment by aaroncraigongithub** [comment]
@pvdlg @travi perhaps I'm configured incorrectly, and npm docs are unclear as to what the setting in package.json should be, but according to this page, I believe I can publish private packages to npm: 
https://docs.npmjs.com/creating-and-publishing-private-packages

I'm assuming by _private_, the docs mean "packages that have `private: true` in their package.json file" , as I am unaware of specifying private packages in any other way.

**Comment by travi** [comment]
the main thing to be aware of is that semantic-release defers to npm itself for that type of config. the npm docs already specify the [behavior of `private`](https://docs.npmjs.com/files/package.json#private). referring to the npm docs is also mentioned in [the readme for this project](https://github.com/semantic-release/npm#npm-configuration).

**Comment by aaroncraigongithub** [comment]
So, `npmPublish` in `.releaserc` can only ever have an effect when it is `false` (in other words, it's solely to be used to disable publishing in a package that would otherwise be published)?

**Comment by pvdlg** [comment]
Yes. That's explained in the docs by mentioning the default value of `npmPublish` is `true`.

Having it also allow users to use the `@sematnic-release/npm` plugin to update their `package.json` without publishing and without having to set `private` to `true`. This useful for Atom packages for example.

The `npmPublish` option was introduced before we started to handle `private`. Looking back we should probably have supported only `private` and find a better way to support the edge case of packages that need to skip publish but can't set `private` to `true`. But now we can't remove the `npmPublish` so easily.

**Comment by aaroncraigongithub** [comment]
Cool makes sense.  I might be useful just to add to the docs something along the lines of ```If your package is set to private in package.json, then this plugin will ignore the `npmPublish` setting altogether. ```   This would mostly be useful to newcomers to the npm publishing pipeline (like me) who may not understand the underlying constraints properly.  Again, happy to open a PR if you feel the docs can be more clear.

**Comment by MRigal** [comment]
I fully agreee with @aaroncraigongithub, just lost more than half an hour because the documentation wording is misleading!

And the use-case of being willing to publish only to a private package registry might justify a line about it in the docs, IMHO
